[MINOR][K8S] add missing docs for podTemplateContainerName properties

aditanase opened the pull request:
## What changes were proposed in this pull request?

Adding docs for an enhancement that came in late in this PR: #22146 
Currently the docs state that we're going to use the first container in a pod template, which was the implementation for some time, until it was improved with 2 new properties.

## How was this patch tested?

I tested that the properties work by combining pod templates with client-mode and a simple pod template.

Please review http://spark.apache.org/contributing.html before opening a pull request.

aditanase committed:
[MINOR][K8S] add missing docs for podTemplateContainerName properties.

AmplabJenkins commented:
Can one of the admins verify this patch?

aditanase commented:
cc @mccheah @onursatici @skonto

AmplabJenkins commented:
Can one of the admins verify this patch?

AmplabJenkins commented:
Can one of the admins verify this patch?

dongjoon-hyun commented:
Thank you for your first contribution, @aditanase .

dongjoon-hyun commented:
ok to test

dongjoon-hyun commented on docs/running-on-kubernetes.md:
```diff
@@ -932,15 +935,31 @@ specific to Spark on Kubernetes.
+  </td>
+</tr>
+<tr>
+  <td><code>spark.kubernetes.driver.podTemplateContainerName</code></td>
```
Could you move this to line 941 after `spark.kubernetes.driver.podTemplateFile` config? We had better group together `spark.kubernetes.driver.*` configs and `spark.kubernetes.executor.*` configs.

aditanase replied:
Thanks, done. I'll squash the commits before merging.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/99341/
Test PASSed.

SparkQA commented:
Kubernetes integration test starting
URL: https://amplab.cs.berkeley.edu/jenkins/job/testing-k8s-prb-make-spark-distribution-unified/5418/

SparkQA commented:
Kubernetes integration test status success
URL: https://amplab.cs.berkeley.edu/jenkins/job/testing-k8s-prb-make-spark-distribution-unified/5418/

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/testing-k8s-prb-make-spark-distribution-unified/5418/
Test PASSed.

SparkQA commented:
**[Test build #99364 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/99364/testReport)** for PR 23155 at commit [`290b764`](https://github.com/apache/spark/commit/290b76443c379e99713be360a176fe771858eade).

SparkQA commented:
**[Test build #99364 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/99364/testReport)** for PR 23155 at commit [`290b764`](https://github.com/apache/spark/commit/290b76443c379e99713be360a176fe771858eade).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/99364/
Test PASSed.

SparkQA commented:
Kubernetes integration test starting
URL: https://amplab.cs.berkeley.edu/jenkins/job/testing-k8s-prb-make-spark-distribution-unified/5441/

SparkQA commented:
Kubernetes integration test status success
URL: https://amplab.cs.berkeley.edu/jenkins/job/testing-k8s-prb-make-spark-distribution-unified/5441/

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/testing-k8s-prb-make-spark-distribution-unified/5441/
Test PASSed.

aditanase commented:
@dongjoon-hyun is this ok to merge? Should I squash it and ping you again?

dongjoon-hyun commented on docs/running-on-kubernetes.md:
```diff
@@ -932,15 +935,31 @@ specific to Spark on Kubernetes.
+  <td><code>spark.kubernetes.driver.podTemplateContainerName</code></td>
+  <td>(none)</td>
+  <td>
+   Specify the container name to be used as a basis for the driver in the given [pod template](#pod-template).
```
Actually, `[pod template](#pod-template)` is not working. Could you try the following and check with `SKIP_API=1 jekyll build`?
```
<a href="#pod-template">pod template</a>
```

dongjoon-hyun commented on docs/running-on-kubernetes.md:
```diff
@@ -932,15 +935,31 @@ specific to Spark on Kubernetes.
   <td>(none)</td>
   <td>
    Specify the local file that contains the driver [pod template](#pod-template). For example
```
Please fix this, too.

dongjoon-hyun commented on docs/running-on-kubernetes.md:
```diff
@@ -932,15 +935,31 @@ specific to Spark on Kubernetes.
   <td><code>spark.kubernetes.executor.podTemplateFile</code></td>
   <td>(none)</td>
   <td>
    Specify the local file that contains the executor [pod template](#pod-template). For example
```
ditto.

dongjoon-hyun commented on docs/running-on-kubernetes.md:
```diff
@@ -932,15 +935,31 @@ specific to Spark on Kubernetes.
+  <td><code>spark.kubernetes.executor.podTemplateContainerName</code></td>
+  <td>(none)</td>
+  <td>
+   Specify the container name to be used as a basis for the executor in the given [pod template](#pod-template).
```
ditto.

srowen commented:
Ping @aditanase

SparkQA commented:
**[Test build #100886 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/100886/testReport)** for PR 23155 at commit [`2f663f1`](https://github.com/apache/spark/commit/2f663f1777521e33512f5b392534b00c0a7f2403).

aditanase commented:
@dongjoon-hyun Thanks for catching the broken anchors. Fixed them per your suggestion and tested with a jekyll build. Should work now.

@srowen my apologies, I've been on an extended winter break. I've rebased against master and squashed my commits - should be ready to merge.

SparkQA commented:
Kubernetes integration test starting
URL: https://amplab.cs.berkeley.edu/jenkins/job/testing-k8s-prb-make-spark-distribution-unified/6778/

SparkQA commented:
**[Test build #100886 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/100886/testReport)** for PR 23155 at commit [`2f663f1`](https://github.com/apache/spark/commit/2f663f1777521e33512f5b392534b00c0a7f2403).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/100886/
Test PASSed.

SparkQA commented:
Kubernetes integration test starting
URL: https://amplab.cs.berkeley.edu/jenkins/job/testing-k8s-prb-make-spark-distribution-unified/6779/

SparkQA commented:
Kubernetes integration test status success
URL: https://amplab.cs.berkeley.edu/jenkins/job/testing-k8s-prb-make-spark-distribution-unified/6778/

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/testing-k8s-prb-make-spark-distribution-unified/6778/
Test PASSed.

SparkQA commented:
Kubernetes integration test status success
URL: https://amplab.cs.berkeley.edu/jenkins/job/testing-k8s-prb-make-spark-distribution-unified/6779/

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/testing-k8s-prb-make-spark-distribution-unified/6779/
Test PASSed.

srowen commented:
Merged to master
